### ISSUE TITLE
SmbFile.listFiles failed on Linux with SMB1 java.net.MalformedURLException: Name must not be empty

### ISSUE THREAD

**Issue by thanh-vn** [AgNO3/jcifs-ng#18]
We are experience another listFiles() issue with SMB1 on Linux to Windows Server 2008 R2.  From our testing, this issue does not happen on Windows (where jcifs-ng is).  It sometimes (about 50%) happens on our Linux environment.  When enabling SMB2, we do not see the issue at all.

2017-11-28 03:42:13 ERROR FileEntryAdapterIterator:? - Failed to create child URL
java.net.MalformedURLException: Name must not be empty
        at jcifs.smb.SmbFile.checkName(Unknown Source)
        at jcifs.smb.SmbFile.<init>(Unknown Source)
        at jcifs.smb.DirFileEntryAdapterIterator.adapt(Unknown Source)
        at jcifs.smb.FileEntryAdapterIterator.advance(Unknown Source)
        at jcifs.smb.FileEntryAdapterIterator.next(Unknown Source)
        at jcifs.smb.FileEntryAdapterIterator.next(Unknown Source)
        at jcifs.smb.SmbEnumerationUtil.listFiles(Unknown Source)
        at jcifs.smb.SmbFile.listFiles(Unknown Source)
        at TestJCIFS_readdir.main(TestJCIFS_readdir.java:39)

I collected network traces below and hope that you can see what may happen with SMB1 case:

smb1_NameMustNotBeEmpty.cap <-- Problem with SMB1 on Linux
smb1_NameMustNotBeEmpty-2ndRun-NoError.cap <-- No Error with SMB1 on Linux
smb2_NameMustNotBeEmpty.cap  <-- No Error with SMB2 on Linux

Thank you very much for your help!

[SMB1-Linux_NameEmpty.zip](https://github.com/AgNO3/jcifs-ng/files/1508461/SMB1-Linux_NameEmpty.zip)

**Comment by mbechler** [comment]
I'm having some trouble reproducing that. I really don't see any meaningful difference between the SMB1 dumps and decoding them seems to work just fine. I'll try some more, but in the meantime can you recheck that these are the correct dump files? It would also be good to know which of the responses is triggering the issue, maybe you can replace the log.error(s)  with throw RuntimeException in FileEntryAdapterIterator so that execution stops when the error occurs?

**Comment by thanh-vn** [comment]
Thanks for the response.  We collected a new network trace.
The problem of empty FileEntry's name during FileEntryAdapterIterator.advance() is not consistent to any folder/file.  However our team saying it is about 50% hit.  The testcase contains two folders: 74 files and 395 files. 
We included here the new dump and hope we can see anything from it.  In the mean time we will try your next suggestion of having log.error(s) and throwing RuntimeException in FileEntryAdapterIterator.
--------------------
2017-11-29 05:25:55 DEBUG FileEntryAdapterIterator:71 - FileEntryAdapterIterator: advance name =tasking letter deliverables 121799-0.xls
2017-11-29 05:25:55 DEBUG FileEntryAdapterIterator:71 - FileEntryAdapterIterator: advance name =
2017-11-29 05:25:55 DEBUG FileEntryAdapterIterator:71 - FileEntryAdapterIterator: advance name =
2017-11-29 05:25:55 ERROR FileEntryAdapterIterator:77 - Failed to create child URL
java.net.MalformedURLException: Name must not be empty
        at jcifs.smb.SmbFile.checkName(SmbFile.java:656)
        at jcifs.smb.SmbFile.<init>(SmbFile.java:533)
        at jcifs.smb.DirFileEntryAdapterIterator.adapt(DirFileEntryAdapterIterator.java:57)
        at jcifs.smb.FileEntryAdapterIterator.advance(FileEntryAdapterIterator.java:74)
        at jcifs.smb.FileEntryAdapterIterator.next(FileEntryAdapterIterator.java:122)
        at jcifs.smb.FileEntryAdapterIterator.next(FileEntryAdapterIterator.java:32)
        at jcifs.smb.SmbEnumerationUtil.listFiles(SmbEnumerationUtil.java:260)
        at jcifs.smb.SmbFile.listFiles(SmbFile.java:1271)
        at TestJCIFS_readdir.main(TestJCIFS_readdir.java:54)
2017-11-29 05:25:55 DEBUG FileEntryAdapterIterator:71 - FileEntryAdapterIterator: advance name =
2017-11-29 05:25:55 ERROR FileEntryAdapterIterator:77 - Failed to create child URL
------------------------------------

Thanks,
[SMB1-Linux-NameMustNotBeEmpty.zip](https://github.com/AgNO3/jcifs-ng/files/1512646/SMB1-Linux-NameMustNotBeEmpty.zip)
